### PR TITLE
Improve MSVC project generation

### PR DESCRIPTION
**Pull request by nOOb3167**
#6835 is two months old. Therefore:
I determine that this code is unmaintained.
I request these needed fixes be merged via special political process.
Maybe someone working on another part of the engine could pretend to have looked at the changes and approve & commit.

This PR begins the resubmission of changes proposed by #6835 part-by-part.

### CODE REVIEW

**Comment by sfan5** [comment]
> CMakeLists pass windows sources (manifest and resource file) to executable targets separately from common_SRCS

why do this?

**Comment by nOOb3167** [comment]
The source_group command's TREE argument mode tries to categorize files by a single path.
Being passed a file with path ${PROJECT_SOURCE_DIR}/somedirectory/somefile.cpp , the source_group command will extract somedirectory/somefile.cpp, and use that for display inside a generated project file.

Two groups of files are to be considered for being passed to the source_group command for this PR:
The resource and manifest files are located outside the ${PROJECT_SOURCE_DIR} path.
Other sources (cpp and h files) are located inside the ${PROJECT_SOURCE_DIR} path.

source_groups limitation to categorize by a single path means there is no way to construct a source_groups call that would accept both of the aforementioned groups.

**Comment by nOOb3167** [comment]
the first commit modifies four lines
the second commit adds four lines

Can i get a short daily status report?

suggested format:
  Status Report Day 48: During the past moon cycle I have reviewed up to but not including the second line of the first commit. However the second lines' sequence of characters between the first opening brace, and the coma, is still proving to be a tough challenge. I however, known and delightful to rise up to said challenge, have laboriously decoded the third and fourth characters of the troublesome sequence (credit goes to 2pac whose music videos have fueled inspiration for today. and of course Spotify for proving to be an invaluable platform for music video delivery). Without further ado let me announce the results: The third character appears to be 'e' and as for the fourth character, it is the evertough cookie also known as 'f'. In the postoperation review I have remarked to the continuous improvements in efficiency and reliability of the review operations. Within the next report I am planning to begin preparation work (estimate 3~ day) on decoding and analyzing characters five to seven. Time accounting was followed strictly and in total I may report, that 3.5 fully compensated work hours have been spent on the Day 48 operation. Please refer to the documentation audit log (google docs link here) for the full details.

Maybe the developers could draw straws as to who gets saddled with this?

**Comment by sfan5** [comment]
not with that attitude

**Comment by nOOb3167** [comment]
What attitude do you prefer then, sir?
#6835 provides detailed description and rationale with screenshots and suchlike.
It is then left to rot.

It is easy to criticize when you do not specify what exactly is wrong with the PRs.
Why are they not being merged or responded to?

The only way for me to know is if someone indeed responds to the PRs but not responding is the problem in the first place thus we end in a Catch-22 situation.

The posts above call for response. I have put the effort into presenting the subject in a humorical matter to avoid offending. But I see it was not appreciated.

**Comment by sfan5** [comment]
> It is easy to criticize when you do not specify what exactly is wrong with the PRs.

I wasn't criticizing the content of your PR, just the comment you made.
> Why are they not being merged or responded to?

This is an open-source project, nobody is paid to work on anything and thus pull requests tend to be left lying around (not saying this is good, but that's how it is).

**Comment by nOOb3167** [comment]
Thank you.
If any issues appear I'll go develop a fix - (Reading through the build system code so far, though, it seems like nothing else is likely to be affected by these changes.)

> This is an open-source project, nobody is paid to work on anything and thus pull requests tend to be left lying around (not saying this is good, but that's how it is).

I hope I can explain / defend from this specific aspect, though:

Notice in the first post the code is called 'unmaintained'.
This occurs naturally not even in resource starved open-source development, even in critical government projects will eventually have people retire, change jobs or simply forget what the code does.
In that case some subsystem will just never get updated again, simply because there is nobody to update it, irregardless of resources available.
For Visual Studio on Minetest, specifically, coming back and compiling Minetest periodically often results in some-or-other breakage, which takes a while to fix. This would indicate that relatively few, if any, core developers are building on that platform (AFAIK cross-compiling with MinGW is the primary way?).
It may be the case that no one is actually there to update the Visual Studio parts of the build system.

The first post then suggests using a 'special political process' - described as maybe having someone working on another part of the engine push the code. This may be the only solution: The usual way where core developers pick up issues may not work - probably nobody feels like they are the right person to review because, for example, the person who originally wrote Visual Studio specific parts of the build system actually is not working on minetest or long time not available.

This call is essentially repeated in the followup comment, as: "Maybe the developers could draw straws as to who gets saddled with this?"
There are two negative parts intentionally inserted: "draw straws" (negative in this context), and "saddled" (negative in pretty much any context).
"Saddled" is used because in the scenario described above, it is accurate: There may be no-one for this specific PR, so if for some reason a developer DID work on the PR, it is probably very unenjoyable. It would be insulting to the developer if I tried to handwave the request, like "if anybody finds time, please throw an eye over here and lend us a hand! :) thanks!", because this PR is basically stinking sewage work.
"Draw straws" is used because this is likely the least unfair way - no singling a specific dev and telling them what to do.

The rest of the comment is basically focused on how the change is relatively trivial (four lines first commit, four lines second), and advertisin how this change is now hopefully easy to review (the spiel absurditizing analysis and decoding of individual characters such as 'e' and 'f', etc).
The original PR (from previous month) actually had an extra part related to recording header files in 
 calls to ADD_EXECUTABLE - which was here cut out for ease of review.

Basically, I did try to be specifically mindful of the nature of open source / volunteer work.

**Comment by celeron55** [comment]
Nice poking there, you know how it is 8)
